cellprofiler/__main__.py

```diff
@@ -576,18 +586,24 @@ def get_batch_commands(filename):
                 print "CellProfiler -c -r -p %s -f %d -l %d" % (filename, prev + 1, off)
 
                 prev = off
-
-            return
-
-    metadata_tags = m.get_grouping_tags()
-
-    groupings = m.get_groupings(metadata_tags)
-
-    for grouping in groupings:
-        group_string = ",".join(["%s=%s" % (k, v) for k, v in grouping[0].iteritems()])
-
-        print "CellProfiler -c -r -p %s -g %s" % (filename, group_string)
-
+    else:
+        metadata_tags = m.get_grouping_tags()
+
+        if len(metadata_tags) == 1 and metadata_tags[0] == 'ImageNumber':
+            #do file numbers
+            for i in range(0, len(image_numbers), n_per_job):
+                first = image_numbers[i]
+                last  = image_numbers[min(i+n_per_job-1, len(image_numbers)-1)]
+                print "CellProfiler -c -r -p %s -f %d -l %d" % (filename, first, last)
+        else:
+            # This is the original code, doesn't work if the only metadata tag is group number
+            # not sure whether it works in other cases.
+            # Also not sure whether this code can even be reached as grouping should land us in the first "if" branch 
```

Comment:
Thanks! I'm going to need some time to verify the behavior, here. Do you mind if I edit your pull request to add automated tests?